src/main/java/de/srendi/advancedperipherals/common/addons/computercraft/apis/EntityAPI.java

```diff
@@ -0,0 +1,81 @@
+package de.srendi.advancedperipherals.common.addons.computercraft.apis;
+
+import dan200.computercraft.api.lua.*;
+import dan200.computercraft.shared.util.NBTUtil;
+import de.srendi.advancedperipherals.common.configuration.APConfig;
+import de.srendi.advancedperipherals.common.util.EntityUtil;
+import de.srendi.advancedperipherals.common.util.LuaConverter;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.player.Player;
+
+import java.util.Map;
+import java.util.UUID;
+
+public class EntityAPI {
+
+    public static final EntityAPI ENTITY_API = new EntityAPI();
+
+    @LuaFunction(mainThread = true)
+    public final Object getNBT(IArguments arguments) throws LuaException {
+        if (!APConfig.API_CONFIG.enableGetNBT.get())
+            throw new LuaException("This function is disabled in the config. Activate it or ask an admin if they can activate it.");
+        Entity entity = EntityUtil.getEntityFromUUID(UUID.fromString(arguments.getString(0)));
+        if (!APConfig.API_CONFIG.enablePlayerAccess.get() && entity instanceof Player)
+            throw new LuaException("Using players in EntityAPI is disabled in the config. Activate it or ask an admin if they can activate it.");
```

Comment:
Do you think that return `nil, error_msg` is better than throw a `LuaException`? Since when the exception happens the client did not pass an unexpected value.
if we returns the error as a return value script may detect that error easier as well.

Reply:
I think that makes sense for `enablePlayerAccess`, as you might be looping through nearby entities calling an EntityAPI function on them and then it would unexpectedly throw if one of those entities is a player. For the enable function config options, if they are disabled the function truly shouldn't be called at all, so I think throwing an error is fine for those. I'll change `enablePlayerAccess` to return `nil, error_msg`.